### ISSUE TITLE
VTK warnings when modifying pipeline

### ISSUE THREAD

**Issue by GaelVaroquaux** [enthought/mayavi#3]
Reported by Maik Beckmann:

This code

<pre>
from enthought.mayavi import mlab

pts = mlab.points3d([0,1], [0,1], [0,1])
pts.mlab_source.set(x=[0,2])
</pre>

works, but gives

<pre>
ERROR: In /build/src/VTK/Filtering/vtkExecutive.cxx, line 756
vtkStreamingDemandDrivenPipeline (0x59b4900): Algorithm
vtkAssignAttribute(0x59bd8a0) returned failure for request:
vtkInformation (0x5ad6c60)
  Debug: Off
  Modified Time: 818743
  Reference Count: 1
  Registered Events: (none)
  Request: REQUEST_DATA_OBJECT
  ALGORITHM_AFTER_FORWARD:
</pre>

where this is the emitting code in vtkExecutive.cxx in
vtkExecutive::CallAlgorithm(...)

<pre>
   ...
  // If the algorithm failed report it now.
  if(!result)
    {
    vtkErrorMacro("Algorithm " << this->Algorithm->GetClassName()
                  << "(" << this->Algorithm
                  << ") returned failure for request: "
                  << *request);
    }
  ...
</pre>

Though its reported as an error by vtk, it doesn't seem to break anything.

**Comment by jessicaabreu** [comment]
+1 for a fix

**Comment by bcorfman** [comment]
+1 for a fix

**Comment by stefanoborini** [comment]
Still present.

**Comment by solarjoe** [comment]
Still present. +1 for a fix.

**Comment by solarjoe** [comment]
I tried a few things and found another workaround.

This will print the error messages:
`plt = mlab.points3d(x, y, z)`

But if you set the forth parameter ("scalars") the error messages are gone.
This works fine for me:
`plt = mlab.points3d(x, y, z, np.ones_like(x))`

The size of the glyphs can be adjusted by using the keywords 'scale_mode' and 'scale_factor'.

**Comment by zhucaoxiang** [comment]
I agree with solarjoe's solution. There might be something wrong with the auto scalars.
